### DISCUSSION TITLE
Displaying text with simple rich text features

### DISCUSSION THREAD

**Discussion by NaLiJa** [mikke89/RmlUi#526] [Q&A] [answered]
Hi there,
looking for some advice. 

What would be the best solution for displaying larger text blocks with some minimal rich text functionality, say  like 

- having chunks with different font formatting and 
- clickable text links which can be handled in lua oder cpp
- only display, no editing

Has anybody perhaps done this with RmlUi before?
Not sure what might be the best way to imlement this. Customize the textarea element? Add a new "label" element that is directly clickable and simply add child elements to the DOM?
Or is this even possible with RmlUi now and I just do not see it?

Thanks for your help!

**Comment by mikke89** [comment, marked as answer]
Maybe I don't fully understand the problem, but as long as you don't need editing, this should be straightforward to do: Just use `<a>` tags, ` <h1>` tags, sprinkle some `<span>`s, and everything else you could need, style them appropriately, add some event handlers, and you should be good to go. This is something the HTML/CSS model excels at in my view. I would stay away from textarea if you don't need text editing.

**Reply by NaLiJa** [reply]
Yes, this is the way. I just wondered if it is the most effective way to implement, but I was probably still thinking too much in traditional UI writing.
